### ISSUE TITLE
Should entity resolvers be limited to the root level.

### ISSUE THREAD

**Issue by michaelstaib** [graphql/composite-schemas-spec#8]
Do we want to allow entity resolvers that can be nested?

```graphql
type Query {
  persons: PersonResolvers! @internal
}

type PersonResolvers @entityResolver {
  byId(id: ID!): Person
  byName(name: String!): Person
}
```

**Comment by martijnwalraven** [comment]
Yes, I think entity resolvers should be allowed on nested objects as long as these are singletons (so as long as the fields that lead to it don't take arguments). Not sure if we should allow `@entityResolver` (or whatever directive we land on) on types however, because the marker applies to individual fields. (Also, I'm assuming `@internal` on `Query.persons` is used here to hide the entity resolvers from the API schema, but you wouldn't generally want that because these fields are useful for clients as well.)

**Comment by PascalSenn** [comment]
We have since agreed on that we can have nested `lookups`

```graphql
type Query {
   workspaceById(id: ID!): Workspace @lookup
}

type Workspace {
   userByName(name: String): User @lookup
}
```
